packages/esm-outpatient-app/src/patient-queue-header/patient-queue-header.component.tsx

```diff
@@ -15,8 +34,40 @@ const PatientQueueHeader: React.FC = () => {
           <p className={styles['productiveHeading04']}>{t('home', 'Home')}</p>
         </div>
       </div>
+      <div className={styles.headerRightContainer}>
+        <div className={styles.locationDateContainer}>
+          <Location16 />
+          <SessionLoction />
+          <span className={styles.dotMark}>.</span>
+          <Calendar16 className={styles.calendar} /> <TodayDate />
+        </div>
+        <div className={styles.dropDownContainer}>
+          <label className={styles.viewLabel}>{t('view', 'View:')} </label>
+          <Dropdown
+            id="typeOfCare"
+            label={t('ncdCare', 'NCD Care')}
+            type="inline"
+            items={items}
+            itemToString={(item) => (item ? item.text : '')}
+          />
+        </div>
+      </div>
     </div>
   );
 };
 
 export default PatientQueueHeader;
+
+function TodayDate() {
+  const today = dayjs(new Date());
+  return <span className={styles.dateLabel}>{today.format('DD  MMMM  YYYY')}</span>;
+}
+
+function SessionLoction() {
+  const userSession = useSessionUser();
+  return userSession ? (
+    <span className={styles.locationLabel}>{userSession.sessionLocation.display}</span>
+  ) : (
+    <span></span>
+  );
+}
```

Comment:
```suggestion
const userSession = useUserSession()
userSession?.sessionLocation?.display
```
you can use optional chaining here

Reply:
Which also means, as I think @donaldkibet is implying, that you can get rid of this component and just inline it.

If you decide to keep it, fix the typo "SessionLoction" --> "SessionLoc**a**tion"

Reply:
@brandones and @donaldkibet  thanks for your reviews, I will act on them appropriately